### ISSUE TITLE
Axios sub-instances

### ISSUE THREAD

**Issue by mhelvens** [axios/axios#1170]
This is a feature request for something I intuitively always thought Axios already had, until I actually tried to use it one day and it didn't work.

```js
// loggedAxios.js
export.defaults = require('axios').create()
loggedAxios.inerceptors.response.use(generateLogMessage)
```
```js
// authorizedAxios.js
export.defaults = require('./loggedAxios').create({
    headers: [{ authorization: `bearer ${token}` }]
})
```
```js
// specificProject.js
const axios = require('./authorizedAxios').create({
    baseUrl: `http://${specificProjectDomain}`
})

await axios.post('/endpoint') // logged, authorized, pointed at project host
```

So basically, I'm asking for `axios.create().create()` to work, applying the expected order of precedence for options and interceptors (see also https://github.com/axios/axios/issues/993). This would be quite useful for large projects inside large companies, where a company-specific axios object could be prepared (logging, doing auth, etc.) that would still work as a drop-in replacement for vanilla axios.

**Comment by heisian** [comment]
The problem is the `utils.merge` function passes by reference on the nested `headers` object. This shouldn't even be called a **feature**, but rather a **bug**. Trying to get #1391 pushed through.

**Comment by emilyemorehouse** [comment]
Closing, as this issue was resolved in #1395. 🙌

**Comment by mhelvens** [comment]
May I ask how this issue was resolved? Even with `axios@0.19.0-beta.1` I am unable to do `axios.create().create()`.

**Comment by xyy94813** [comment]
why not to support a new api `instance.clone()`?

**Comment by iamgreyy** [comment]
Why did this issue is closed? It is not working in 0.19b.
